move-to-sdk/app/io/sphere/sdk/play/metrics/MetricAction.java

```diff
@@ -44,6 +45,7 @@ private void logRequestDataOnComplete(final Http.Context ctx, final List<ReportR
     }
 
     private Integer calculateTotalSize(final List<ReportRawData> rawDatas) {
-        return rawDatas.stream().map(elem -> elem.getHttpResponse().getResponseBody().map(b -> b.length).orElse(0)).reduce(0, (a, b) -> a + b);
+        return rawDatas.stream().map(elem -> Optional.ofNullable(elem.getHttpResponse().getResponseBody())
+                .map(b -> b.length).orElse(0)).reduce(0, (a, b) -> a + b);
```

Comment:
Not completely sure what you want to do, so I think it's better if you do it yourself. I just touched this line of code because of M16 and it was you who implemented it initially, so you can improve it anytime, just go ahead in a PR :)